client/src/controllers/TabsController.ts

```diff
@@ -0,0 +1,319 @@
+/* eslint-disable no-shadow */
+import { Controller } from '@hotwired/stimulus';
+import { debounce } from '../utils/debounce';
+
+interface IndexedEventTarget extends EventTarget {
+  index: number;
+}
+
+interface TabLink extends HTMLAnchorElement {
+  index: number;
+}
+
+/**
+ * @example - creating a simple tabs interface
+ *<div class="w-tabs" data-controller="w-tabs" data-action="popstate@window->w-tabs#loadHistory" data-w-tabs-selected-value="" data-w-tabs-selected-class="animate-in">
+ *    <div class="w-tabs__list" role="tablist" data-w-tabs-target="list">
+ *        <a id="tab-label-tab-1" href="#tab-tab-1" class="w-tabs__tab" role="tab" tabindex="-1"
+ *           data-action="click->w-tabs#handleTabChange:prevent keydown.right->w-tabs#selectNext keydown.left->w-tabs#selectPrevious keydown.home->w-tabs#selectFirst keydown.end->w-tabs#selectLast" data-w-tabs-target="label">
+ *            Tab 1
+ *        </a>
+ *        <a id="tab-label-tab-2" href="#tab-tab-2" class="w-tabs__tab" role="tab"
+ *           data-action="click->w-tabs#handleTabChange:prevent keydown.right->w-tabs#selectNext keydown.left->w-tabs#selectPrevious keydown.home->w-tabs#selectFirst keydown.end->w-tabs#selectLast" data-w-tabs-target="label">
+ *            Tab 2
+ *        </a>
+ *    </div>
+ *    <div class="tab-content tab-content--comments-enabled">
+ *        <section id="tab-tab-1" class="w-tabs__panel " role="tabpanel" aria-labelledby="tab-label-tab-1" data-w-tabs-target="panel">
+ *            tab-1
+ *        </section>
+ *        <section id="tab-tab-2" class="w-tabs__panel " role="tabpanel" aria-labelledby="tab-label-tab-2" data-w-tabs-target="panel">
+ *            tab-2
+ *        </section>
+ *    </div>
+ *</div>
+ */
+
+export class TabsController extends Controller<HTMLDivElement> {
+  static targets = ['list', 'trigger', 'label', 'panel'];
```

Comment:
```suggestion
  static targets = ['label', 'panel', 'trigger'];
```

Let's order alphabetically and as noted in other comments we can probably get rid of the `list` target for simplicity.